numpy/core/_methods.py

```diff
@@ -55,6 +55,18 @@ def _count_reduce_items(arr, axis):
         items *= arr.shape[ax]
     return items
 
+def _clip(a, min=None, max=None, out=None):
+    casting = 'unsafe'  # TODO: probably harmful, but backwards-compatible
+
+    if min is None and max is None:
+        raise ValueError("One of max or min must be given")
+    elif min is None:
+        return um.clip_above(a, max, out=out, casting=casting)
```

Comment:
Can we not use `um.minimum` here? (and `um.maximum` instead of `clip_below`)

Reply:
Nope - clip needs to preserve NaNs in and only in the first argument. maximum always propagates, and fmax always ignores. So we need a third maximum-like function